src/meta/src/hummock/manager/checkpoint.rs

```diff
@@ -119,18 +126,29 @@ impl HummockManager {
         let timer = self.metrics.version_checkpoint_latency.start_timer();
         // 1. hold read lock and create new checkpoint
         let versioning_guard = read_lock!(self, versioning).await;
-        let versioning = versioning_guard.deref();
-        let current_version = &versioning.current_version;
-        let old_checkpoint = &versioning.checkpoint;
+        let versioning: &Versioning = versioning_guard.deref();
+        let current_version: &HummockVersion = &versioning.current_version;
+        let old_checkpoint: &HummockVersionCheckpoint = &versioning.checkpoint;
         let new_checkpoint_id = current_version.id;
         let old_checkpoint_id = old_checkpoint.version.id;
         if new_checkpoint_id < old_checkpoint_id + min_delta_log_num {
             return Ok(0);
         }
+        if cfg!(test) && new_checkpoint_id == old_checkpoint_id {
+            drop(versioning_guard);
+            let mut versioning = write_lock!(self, versioning).await;
+            versioning.mark_objects_for_deletion();
+            let min_pinned_version_id = versioning.min_pinned_version_id();
+            trigger_gc_stat(&self.metrics, &versioning.checkpoint, min_pinned_version_id);
+            return Ok(0);
+        }
```

Comment:
Why do we need this specific logic only for test?

Reply:
This branches only happens when `min_delta_log_num` is 0. In usual deployment we should have set it to a non-zero value, and then we won't reach this branch. Only in test we will pass the param as 0 to trigger a deletion and gc stats.